addons/mail/models/models.py

```diff
@@ -246,3 +247,108 @@ def _notify_email_header_dict(self):
         return {
             'X-Odoo-Objects': "%s-%s" % (self._name, self.id),
         }
+
+    def _get_safe_create_data(self, vals_list, is_remove_missing_ref=False):
+        """ Return cleaned create vals_list to avoid missing reference, creation of new record, deletion and update by:
+
+        - Removing set to any non-existing id for relations many2one and many2many if is_remove_missing_ref
+        - Removing set to non-existing fields
+        - Removing set to one2many relation (which will update the other side of the relation)
+        - Removing set that create sub-record
+        - Removing any command != LINK or SET (i.e. skipping CREATE, UPDATE, DELETE, UNLINK, CLEAR)
+
+        :param list|dict vals_list: value to be cleaned
+        :param bool is_remove_missing_ref: whether to remove missing reference or not (see above). If set to False, no
+        DB query is done and the method is fast.
+        """
+        if isinstance(vals_list, dict):
+            vals_list = [vals_list]
+
+        errors = defaultdict(list)
+
+        def filter_existing_ids(f_name, ids):
+            """ Filter existing ids if is_remove_missing_ref, otherwise return the ids as is.
+
+             :param str f_name: field name of this record
+             :param list ids: id of the target of the field relation
+             :return: If is_remove_missing_ref, returns only existing ids of the relation defined by f_name
+             (being active or not) otherwise returns the ids as is.
+             Note: Null (False, None, ...) are considered as existing ids.
+             """
+            if not is_remove_missing_ref:
+                return ids
+
+            def not_null(c_id):
+                """Check if the id is not null in the DB senses"""
+                return c_id is not False and c_id is not None  # touchy because False is an int and False == 0!
```

Comment:
I am surprised you didn't make it more complicated :smile: 
```suggestion
                return bool(c_id)
```